python/mxnet/ndarray/random.py

```diff
@@ -518,3 +518,47 @@ def shuffle(data, **kwargs):
     <NDArray 2x3 @cpu(0)>
     """
     return _internal._shuffle(data, **kwargs)
+
+
+def randint(low=0, high=1, shape=_Null, dtype=_Null, ctx=None, out=None, **kwargs):
+    """Draw random samples from a discrete uniform distribution.
+
+    Samples are uniformly distributed over the half-open interval *[low, high)*
+    (includes *low*, but excludes *high*).
+
+    Parameters
+    ----------
+    low : float or NDArray
+        Lower boundary of the output interval. All values generated will be
+        greater than or equal to low. The default value is 0.
+    high : float or NDArray
+        Upper boundary of the output interval. All values generated will be
+        less than high. The default value is 1.
+    shape : int or tuple of ints
+        The number of samples to draw. If shape is, e.g., `(m, n)` and `low` and
+        `high` are scalars, output shape will be `(m, n)`. If `low` and `high`
+        are NDArrays with shape, e.g., `(x, y)`, then output will have shape
+        `(x, y, m, n)`, where `m*n` samples are drawn for each `[low, high)` pair.
```

Comment:
this behavior of the `randint` operator seems weird. 

Why are we not mimicking the behavior of [`numpy.random.randint`](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.random.randint.html) function. Even pytorch's [`torch.randperm`](https://pytorch.org/docs/0.3.1/torch.html?highlight=torch%20random#torch.randperm) is defined similarly.

Reply:
I have tried to follow the process of sampling used in other Random generators in MXNet to maintain consistency. 
For instance, you can find similar departure in convention for # of sample to draw in random.uniform (or others) of MXNet vis-a-vis Numpy

Reply:
are other random samplers a departure from convention - https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.uniform.html

Reply:
As discussed in this [dev thread](https://lists.apache.org/thread.html/1e80983b9e78b88dc5e5d83edadc0f86b9e368702b954ce05d212e3c@%3Cdev.mxnet.apache.org%3E) there are two sets of sampling functions within MXNet `sample_x` and `random_x`. make sure your implementation is consistent with this and update the documentation accordingly.